src/hooks/usePageReducer.js

```diff
@@ -32,22 +37,27 @@ const usePageReducer = (
   debounceTimeout = 250,
   instantDebounceTimeout = 250
 ) => {
-  const columns = getColumns(page);
-  const pageInfo = getPageInfo(page);
+  const columns = useMemo(() => getColumns(page), [page]);
+  const pageInfo = useMemo(() => getPageInfo(page), [page]);
   const memoizedReducer = useMemo(() => getReducer(page), []);
   const [state, dispatch] = useReducer(memoizedReducer, {
     ...initialState,
     columns,
     pageInfo,
   });
 
-  const debouncedDispatch = useCallback(debounce(dispatch, debounceTimeout), []);
+  const thunkDispatcher = action => {
+    if (typeof action === 'function') action(dispatch, state);
+    dispatch(action);
```

Comment:
I'd have thought this line would have to be in the `else` clause of the above `if`? Dispatch is receiving a function as an arg otherwise.

Reply:
oopsies d9b0921ca870399557e490400cb446f387f8a742